rinja_derive/src/generator.rs

```diff
@@ -104,6 +104,16 @@ impl<'a> Generator<'a> {
         Ok(buf.buf)
     }
 
+    fn push_locals<T, F: FnMut(&mut Self) -> Result<T, CompileError>>(
```

Comment:
Can be `FnOnce`, so the wrapped code knows that it may "use up" any variables in its closure.

Reply:
Good catch!